### ISSUE TITLE
Request CTO validation is not working

### ISSUE THREAD

**Issue by dselman** [accordproject/concerto#157]
**Describe the bug**
It is possible to send an invalid request (with respect to the CTO model) and have the Ergo execute.

**To Reproduce**
1. Download the attached template
2. Using the request and state within the archive
3. `cicero execute --request monthlysales-invalid.json --state state.json`
4. Note that execution succeeds, when it should fail

**Expected behavior**
The execution should fail because `sales[0].price.doubleValue` is a string.

**Version**

Cicero 0.13.5

**Comment by jeromesimeon** [comment]
Concerto is generally quite lose with validation. Bug or feature?

```
bash-3.2$ cat models/model.cto 
namespace org.accordproject.promissorynote

import org.accordproject.money.MonetaryAmount from https://models.accordproject.org/money.cto

transaction Payment extends Request {
  o MonetaryAmount amountPaid
  o Boolean isValid
	o Integer someNumber
}
bash-3.2$ cat request.json 
{
  "$class": "org.accordproject.promissorynote.Payment",
  "amountPaid": {
    "doubleValue": 1.0,
    "currencyCode": "USD"
  },
  "isValid" : "something",
  "someNumber" : "something else"
}
bash-3.2$ concerto validate --ctoFiles models/*.cto --sample request.json 
8:45:45 AM - info:
{
  "$class": "org.accordproject.promissorynote.Payment",
  "amountPaid": {
    "$class": "org.accordproject.money.MonetaryAmount",
    "doubleValue": 1,
    "currencyCode": "USD"
  },
  "isValid": false,
  "someNumber": null,
  "transactionId": "7a521051-6af6-4aca-8a26-199f66b09753",
  "timestamp": "2019-12-06T13:45:45.698Z"
}
```

**Comment by jeromesimeon** [comment]
A proposed resolution is to change the validation semantics to be strict. This means atomic values in the JSON will have to be proper instances of the corresponding atomic type.

## Before

### Boolean
```
true validate Boolean -> true
false validate Boolean -> false
"true" validate Boolean -> true
3.14 validate Boolean -> false
null validate Boolean -> false
undefined validate Boolean -> false
{} validate Boolean -> false
[] validate Boolean -> false
```

## After

### Boolean
```
true validate Boolean -> true
false validate Boolean -> false
"true" validate Boolean -> ERROR
3.14 validate Boolean -> ERROR
null validate Boolean -> ERROR
undefined validate Boolean -> ERROR
{} validate Boolean -> ERROR
```

**Comment by jeromesimeon** [comment]
Initial PR implementing the proposal is in #161

**Comment by jeromesimeon** [comment]
> Initial PR implementing the proposal is in #161

The change in this PR is great at identifying potential issues in test. E.g., in Ergo:

https://github.com/accordproject/ergo/pull/724

**Comment by jeromesimeon** [comment]
Fixed based on proposal in #161 in the `1.0` branch
